lib/engine/game/g_18_india/step/certificate_selection.rb

```diff
@@ -55,57 +58,75 @@ def players_visible?
             false
           end
 
+          def show_companies
+            true
+          end
+
+          def show_map
+            true
+          end
+
           def name
-            'Hand Selection'
+            'Initial Hand Selection'
           end
 
           def description
             "Select #{@cards_to_keep} Certificates for your starting hand"
           end
 
-          def finished?
-            @confirmed_selections == @game.players.size
+          def selection_note
+            [
+              "Made #{number_of_selections} of #{@cards_to_keep} selections.",
+              'Click on card to select or unselect it.',
+            ]
           end
 
-          def selections_completed?
-            number_of_selections == @cards_to_keep
+          def select_company(player, company)
+            return if selections_completed? && company.owner.nil? # prevent selecting more than allowed
+
+            # add or remove from choices
+            if company_selected?(company)
+              @choices[player].delete(company)
+              company.owner = nil
```

Comment:
You're setting company ownership in `process_select_multiple_companies`. Does this need to be done here as well?

Reply:
It doesn't NEED to be done here as well. Originally, changing ownership was the only visual cue that the player has selected that company card.  With this updates, there are three indications: the blue highlight, the owner change, and the list of companies on the player card.  My take was the extra visual cue didn't hurt, so I left it there.  The "real" change in ownership happens in `process_select_multiple_companies`, as that is the only method run when the game is loading.